src/lib.rs

```diff
@@ -491,3 +459,45 @@ impl SP3 {
             .flat_map(|(e, sv)| sv.iter().map(|(sv, clk)| (*e, *sv, *clk)))
     }
 }
+
+#[cfg(feature = "nyx-space")]
+use nyx_space::cosmic::{Frame, Orbit};
+
+#[cfg(feature = "nyx-space")]
+impl SP3 {
+    /// Returns Reference Frame used for all coordinates in this file
+    pub fn reference_frame(&self) -> Frame {
+        match self.coord_system.as_str() {
+            "ITR91" | "IGS05" | "IGS20" | "IGb00" | "IGS14" | "ITRF2" => {
```

Comment:
@ChristopherRabotin,

are you aware of any existing Rust library to manage Reference Frames correctly ? 
I don't think such thing exists.
I don't think it belongs in this crate either, `nyx` would be a better place.

Reply:
I'm sorry but I can't unterstand what can be achieved with Nyx. I saved the previous state in a `nyx` branch, we'll see later.
It is much easier for me to implement the lagrangian calculation myself and get started